<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns2="http://www.tibco.com/CommonLE2/namespace/Administration/GUIServices/Category" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns1="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:ns3="http://www.tibco.com/CommonLE2/Public/Common" xmlns:pfx1="http://www.tibco.com/CommonLE2/Public/Common/Dummy" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsdlPfx="http://www.tibco.com/CommonLE2/namespace/SecurityManagement/GUIServices" xmlns:pfx5="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/SecurityConfiguration/AllRolesRetrievalResponse" xmlns:pfx2="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:ns12="http://www.tibco.com/CommonLE/namespace/Common">
    <wsdl:import namespace="http://www.tibco.com/CommonLE2/namespace/SecurityManagement/GUIServices" location="/SharedResources/WSDL/GUI/SecurityManagement/intfUpdateRoleACL.wsdl"/>
    <pd:name>SecurityManagement/GUIServices/wsAvailableRolesRetrieval.process</pd:name>
    <pd:startName>Start</pd:startName>
    <pd:startType wsMsgRef="wsdlPfx:Dummy"/>
    <pd:startX>145</pd:startX>
    <pd:startY>43</pd:startY>
    <pd:returnBindings>
        <wsdlPfx:AllRoles>
            <parameters>
                <xsl:copy-of select="$Call-Process/pfx5:AllRoles"/>
            </parameters>
        </wsdlPfx:AllRoles>
    </pd:returnBindings>
    <pd:endName>End</pd:endName>
    <pd:endType wsMsgRef="wsdlPfx:AllRoles"/>
    <pd:endX>450</pd:endX>
    <pd:endY>43</pd:endY>
    <pd:errorSchemas>
        <Fault wsMsgRef="wsdlPfx:Error"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1285919747863</pd:targetNamespace>
    <pd:activity name="Admin Connection Error">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>145</pd:x>
        <pd:y>166</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=AdministratorConnectError namespace=http://www.tibco.com/CommonLE2/namespace/private/Common</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Call Process">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>299</pd:x>
        <pd:y>45</pd:y>
        <config>
            <processName>/SecurityManagement/DataAccess/Common/AvailableRolesRetrieval.process</processName>
        </config>
        <pd:inputBindings>
            <xsl:copy-of select="$Start/wsdlPfx:Dummy4/parameters/pfx1:Dummy"/>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch All Errors">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>145</pd:x>
        <pd:y>263</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>true</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Generate Internal Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>450</pd:x>
        <pd:y>275</pd:y>
        <config>
            <faultName>Fault</faultName>
        </config>
        <pd:inputBindings>
            <ns1:ActivityInput>
                <xsl:if test="$_error/ns:ErrorReport/Msg">
                    <message>
                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                    </message>
                </xsl:if>
                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                    <messageCode>
                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                    </messageCode>
                </xsl:if>
                <data>
                    <wsdlPfx:Error>
                        <Error>
                            <ns3:Error>
                                <xsl:if test="$_error/ns:ErrorReport/Msg">
                                    <ns3:Message>
                                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                                    </ns3:Message>
                                </xsl:if>
                                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                                    <ns3:Code>
                                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                                    </ns3:Code>
                                </xsl:if>
                            </ns3:Error>
                        </Error>
                    </wsdlPfx:Error>
                </data>
            </ns1:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Generate Record Not Found Fault">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>300</pd:x>
        <pd:y>167</pd:y>
        <config>
            <faultName>Fault</faultName>
        </config>
        <pd:inputBindings>
            <ns1:ActivityInput>
                <message>
                    <xsl:value-of select="$Admin-Connection-Error/pfx2:AdministratorConnectError/pfx2:Message"/>
                </message>
                <messageCode>
                    <xsl:value-of select="$Admin-Connection-Error/pfx2:AdministratorConnectError/pfx2:Code"/>
                </messageCode>
                <data>
                    <wsdlPfx:Error>
                        <Error>
                            <ns3:Error>
                                <ns3:Message>
                                    <xsl:value-of select="$Admin-Connection-Error/pfx2:AdministratorConnectError/pfx2:Message"/>
                                </ns3:Message>
                                <ns3:Code>
                                    <xsl:value-of select="$Admin-Connection-Error/pfx2:AdministratorConnectError/pfx2:Code"/>
                                </ns3:Code>
                            </ns3:Error>
                        </Error>
                    </wsdlPfx:Error>
                </data>
            </ns1:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Local Exception Handler">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>300</pd:x>
        <pd:y>273</pd:y>
        <config>
            <processName>/Common/Local Exception Handler.process</processName>
        </config>
        <pd:inputBindings>
            <ns12:Exception>
                <ErrorMessage>
                    <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                </ErrorMessage>
                <ErrorCode>
                    <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                </ErrorCode>
                <ns12:Error>
                    <xsl:for-each xmlns:xsl="http://www.w3.org/1999/XSL/Transform" select="$_error/ns:ErrorReport">
                        <ns12:ErrorReport>
                            <StackTrace>
                                <xsl:value-of select="StackTrace"/>
                            </StackTrace>
                            <Msg>
                                <xsl:value-of select="Msg"/>
                            </Msg>
                            <FullClass>
                                <xsl:value-of select="FullClass"/>
                            </FullClass>
                            <Class>
                                <xsl:value-of select="Class"/>
                            </Class>
                            <ProcessStack>
                                <xsl:value-of select="ProcessStack"/>
                            </ProcessStack>
                            <MsgCode>
                                <xsl:value-of select="MsgCode"/>
                            </MsgCode>
                        </ns12:ErrorReport>
                    </xsl:for-each>
                </ns12:Error>
            </ns12:Exception>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Call Process</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Call Process</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Admin Connection Error</pd:from>
        <pd:to>Generate Record Not Found Fault</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch All Errors</pd:from>
        <pd:to>Local Exception Handler</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Local Exception Handler</pd:from>
        <pd:to>Generate Internal Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>